web_server/registration.py

```diff
@@ -16,14 +17,19 @@ def registration():
                 error = 'enter {}'.format(parameter)
                 return render_template('registration.html', error=error)
 
-            if parameter == "name":
-                users[request.form.get("name")] = {}
-            elif parameter != "name":
-                users[request.form.get("name")][parameter] = request.form.get(parameter)
-        print(users)
-        return redirect(url_for('map.map'))
-
-    return render_template("registration.html", error=False)
+        if request.form.get("password1") != request.form.get("password2"):
+            error = 'passwords do not match'
+            return render_template('registration.html', error=error)
 
+        user_dict = {'email': request.form.get("email"),
+                     'password': request.form.get("password1"),
+                     'username': request.form.get("name")}
```

Comment:
Maybe we should check if the email is written in a valid format (so that later we can verify it).